app/lib/search/updater.dart

```diff
@@ -106,6 +106,9 @@ class BatchIndexUpdater implements TaskRunner {
     }
     if (_batch.isEmpty) return;
 
+    final Timer timer = new Timer(const Duration(minutes: 5), () {
+      _logger.severe('Batch updated failed to complete in 5 minutes.');
```

Comment:
Do we want to complete the future here as well, possibly with an error? Or would that not help anything?

Reply:
First I'd like to see if this is really the case. Completing with an error would make sense, but would introduce additional error handling too. Let's see the logs first.